tests/Unit/Domain/Services/TalkRating/TalkRatingContextTest.php

```diff
@@ -23,53 +25,49 @@
 class TalkRatingContextTest extends \PHPUnit\Framework\TestCase
 {
     /**
-     * @test
+     * @dataProvider strategyProvider
      */
-    public function getTalkStrategyReturnsATalkRatingStrategy()
+    public function testGetTalkRatingStrategyReturnsCorrectInstance($input, $expectedClassName)
     {
-        $strategy = TalkRatingContext::getTalkStrategy('yesno', $this->authMock());
-        $this->assertInstanceOf(TalkRatingStrategy::class, $strategy);
+        $strategy = TalkRatingContext::getTalkStrategy($input, $this->authMock());
+        $this->assertInstanceOf($expectedClassName, $strategy);
     }
 
-    /**
-     * @test
-     */
-    public function getTalkStrategyWithYesNoReturnsYesNoRating()
+    public function strategyProvider(): array
     {
-        $strategy = TalkRatingContext::getTalkStrategy('yesno', $this->authMock());
-        $this->assertInstanceOf(YesNoRating::class, $strategy);
+        return [
+            'Input is empty string' =>
+                ['', YesNoRating::class],
+            'Input is yesno' =>
+                ['yesno', YesNoRating::class],
+            'Input is onetoten' =>
+                ['onetoten', OneToTenRating::class],
+            'Casing of yesno doesnt matter' =>
+                ['YeSNo', YesNoRating::class],
+            'Casing of onetoten doesnt matter' =>
+                ['OnEToTen', OneToTenRating::class],
+            'Giberish defaults to YesNo' =>
+                ['asdfgo87yhl', YesNoRating::class],
+        ];
     }
 
-    /**
-     * @test
-     */
-    public function casingDoesNotMatterForGetTalkRatingStrategy()
+    public function testNullInputThrowsAnError()
     {
-        $strategy = TalkRatingContext::getTalkStrategy('YeSNo', $this->authMock());
-        $this->assertInstanceOf(YesNoRating::class, $strategy);
+        //This means null input should be checked before handing it to the getTalkStrategy function
+        $this->expectException(\TypeError::class);
```

Comment:
Not sure if we should test this, if we wanted to do it consistently, we would have to test this everywhere in the code base where we use type declarations, and I believe one of the big advantages of using type declarations is that we don't.

What do you think?